src/Symfony/Bundle/FrameworkBundle/Logger/Logger.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Logger;
+
+use Psr\Log\AbstractLogger;
+use Psr\Log\InvalidArgumentException;
+use Psr\Log\LogLevel;
+
+/**
+ * Minimalist PSR-3 logger designed to write in stdout, stderr or any other stream.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ *
+ * @see http://www.php-fig.org/psr/psr-3/
+ */
+final class Logger extends AbstractLogger
+{
+    /**
+     * @var array
+     */
+    private static $defaultLevelToStream = array(
+        LogLevel::EMERGENCY => 'php://stderr',
+        LogLevel::ALERT => 'php://stderr',
+        LogLevel::CRITICAL => 'php://stderr',
+        LogLevel::ERROR => 'php://stderr',
+        LogLevel::WARNING => 'php://stdout',
+        LogLevel::NOTICE => 'php://stdout',
+        LogLevel::INFO => 'php://stdout',
+        LogLevel::DEBUG => false,
+    );
+
+    private $levelToStream;
+    private $formatPattern;
+
+    /**
+     * @var bool
+     */
+    private $errored = false;
+
+    /**
+     * @param array  $levelToStream
+     * @param string $formatPattern
+     */
+    public function __construct(array $levelToStream = array(), $formatPattern = '[%s] %s')
+    {
+        $this->levelToStream = $levelToStream + self::$defaultLevelToStream;
+        $this->formatPattern = $formatPattern;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function log($level, $message, array $context = array())
+    {
+        if (!isset($this->levelToStream[$level])) {
+            throw new InvalidArgumentException(sprintf('The log level "%s" does not exist.', $level));
+        }
+
+        if (!$this->errored) {
+            $this->errored = in_array($level, array(LogLevel::EMERGENCY, LogLevel::ALERT, LogLevel::CRITICAL, LogLevel::ERROR));
+        }
+
+        if ($this->levelToStream[$level]) {
+            file_put_contents($this->levelToStream[$level], sprintf($this->formatPattern, $level, $this->interpolate($message, $context)).PHP_EOL, FILE_APPEND);
+        }
+    }
+
+    /**
+     * Returns true when any messages have been logged at error levels.
+     *
+     * @return bool
+     */
+    public function hasErrored()
```

Comment:
not sure we need the method?

Reply:
In `ConsoleLogger`, it has been introduced by Nicolas in #19090. The use case is to be able to detect if an error has occurred and to set the appropriate exit code. But it doesn't look to be used in the Symfony code base and the PR doesn't give more information.
I've added this for the sake of consistency but maybe should it be removed?

Reply:
OK, let's keep it :)

Reply:
Ok, can you give me a bit more information about the use case? (this method isn't in any interface)

Reply:
I dig a bit and we used to use it in the Blackfire Player - we don't anymore.
So we can consider it a non-common case and remove it from the simple logger we want here.
Let's drop :)